src/commands/project/pipeline.ts

```diff
@@ -18,252 +24,180 @@ import {
-    );
-  }
-  if (typeof orgName !== "string") {
-    missingConfig.push(
-      `--org-url must be of type 'string', ${typeof orgName} given.`
-    );
-  }
-  if (typeof repoName !== "string") {
-    missingConfig.push(
-      `--repo-name must be of type 'string', ${typeof repoName} given.`
-    );
-  }
-  if (typeof repoUrl !== "string") {
-    missingConfig.push(
-      `--repo-url must be of type 'string', ${typeof repoUrl} given.`
-    );
-  }
-  if (typeof devopsProject !== "string") {
-    missingConfig.push(
-      `--devops-project must be of type 'string', ${typeof devopsProject} given.`
-    );
-  }
-  if (typeof buildScriptUrl !== "string") {
-    missingConfig.push(
-      `--build-script-url must be of type 'string', ${typeof buildScriptUrl} given.`
-    );
+export const fetchValidateValues = (
+  opts: ICommandOptions,
+  gitOriginUrl: string,
+  spkConfig: IConfigYaml | undefined
+): ICommandOptions | null => {
+  if (!spkConfig) {
+    throw new Error("SPK Config is missing");
   }
+  const azure_devops = spkConfig?.azure_devops;
+
+  const values: ICommandOptions = {
+    buildScriptUrl: opts.buildScriptUrl || BUILD_SCRIPT_URL,
+    devopsProject: opts.devopsProject || azure_devops?.project,
+    orgName: opts.orgName || azure_devops?.org,
+    personalAccessToken: opts.personalAccessToken || azure_devops?.access_token,
+    pipelineName:
+      opts.pipelineName || getRepositoryName(gitOriginUrl) + "-lifecycle",
+    repoName: opts.repoName || getRepositoryName(gitOriginUrl),
+    repoUrl: opts.repoUrl || getRepositoryUrl(gitOriginUrl)
+  };
 
-  if (missingConfig.length > 0) {
-    logger.error("Error in configuration: " + missingConfig.join(" "));
-    return false;
-  }
+  const map: { [key: string]: string | undefined } = {};
+  (Object.keys(values) as Array<keyof ICommandOptions>).forEach(key => {
+    const val = values[key];
+    if (key === "personalAccessToken") {
+      logger.debug(`${key}: XXXXXXXXXXXXXXXXX`);
+    } else {
+      logger.debug(`${key}: ${val}`);
+    }
+    map[key] = val;
+  });
 
-  return true;
+  const error = validateForRequiredValues(decorator, map);
+  return error.length > 0 ? null : values;
 };
 
 /**
- * Install the project hld lifecycle pipeline in an azure devops org.
+ * Executes the command.
  *
- * @param orgName
- * @param personalAccessToken
- * @param pipelineName
- * @param repositoryName
- * @param repositoryUrl
- * @param project
- * @param buildScriptUrl Build Script URL
- * @param exitFn
+ * @param opts Options object from commander.
+ * @param projectPath Project path which is the current directory.
+ * @param exitFn Exit function.
  */
-export const installLifecyclePipeline = async (
-  orgName: string,
-  personalAccessToken: string,
-  pipelineName: string,
-  repositoryName: string,
-  repositoryUrl: string,
-  project: string,
-  buildScriptUrl: string,
-  exitFn: (status: number) => void
+export const execute = async (
```

Comment:
Can this method name be more descriptive? ie. `executeInstallPipelineCommand`?

Reply:
the goal is the standard all command related file. that is it shall have a
1. commandDecorator function
2. execute function.

Once we standardize all these files, we can use reflective to get the required function and execute them accordingly.

Reply:
Okay, that's fine.

Reply:
thanks